core/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -678,10 +679,27 @@ public void postIndex(Engine.Index index, Throwable ex) {
         try {
             shard.index(index);
             fail();
-        }catch (IllegalIndexShardStateException e){
+        } catch (IllegalIndexShardStateException e) {
 
         }
 
         assertTrue(postIndexWithExceptionCalled.get());
     }
+
+    /**
+     * We currently start check for time/size/ops based tanslog flush as soon as the shard is created. At that time the
+     * engine is not yet started. We should properly deal with this and schedule a future check
+     */
+    public void testFlushRescheduleOnEngineNotAvailable() throws IOException {
+        createIndex("test");
+        ensureGreen();
+        IndicesService indicesService = getInstanceFromNode(IndicesService.class);
+        IndexService test = indicesService.indexService("test");
+        IndexShard shard = test.shard(0);
+        shard.close("Unexpected close", false);
+        shard.state = IndexShardState.CREATED;
```

Comment:
yeah, different kind of evil. I stole this one from another test here. Can try.. not sure if it adds much - except maybe that we can check more states like before the recovering call and also store/snapshot recovery etc.  Tried to keep it simple :)